File: ira/uiserializer/src/test/java/uiserializer/bus/BusTest.java
package uiserializer.bus;

import static org.junit.Assert.assertEquals;
import static uiserializer.bus.NaronEngine.newJSIntance;

import java.io.FileReader;
import java.io.Reader;

import javax.script.ScriptEngine;
import javax.script.ScriptException;

import org.junit.Before;
import org.junit.Test;

public class BusTest {

	private final ScriptEngine engine = newJSIntance();

	@Before
	public final void setUpBusTest() throws Exception {
		try (Reader reader = new FileReader("scripts.js")) {
			engine.eval(reader);
		}
	}

	@Test
	public void test$001() throws Exception {
		final StringBuilder script = new StringBuilder();

		script.append("var marker = undefined;");
		script.append("bus.subscribe('theme1', function(e){ marker = e ; });");
		script.append("bus.publish('theme1','someValue');");
		script.append("marker;");
		assertEquals("someValue", engine.eval(script.toString()));
	}

	@Test(expected = ScriptException.class)
	public void test$002() throws Exception {
		final StringBuilder script = new StringBuilder();
		script.append("marker;");
		engine.eval(script.toString());
	}

}

File: ira/uiserializer/scripts/busTest.js
"bus.js"

test("called", function() {
	// var called = "not called";
	//
	// bus.subscribe("queue", function() {
	// called = true;
	// });
	//
	// bus.publish("queue", {});
	//
	// assertEquals(true, called);

});

test("test queue creating", function() {
	var qName = "queue2";
	try {

		bus.publish(qName, {});
	} catch (e) {
		assertEquals(qName + " not exists", e.message);
		return;
	}
	assertFailure("Expected exception not throwed by publishing in to " + qName);
});